tests/testthat/test-proportional_hazards-glmnet.R

```diff
@@ -149,20 +173,55 @@ test_that("survival probabilities - non-stratified model", {
     NA
   )
 
-  pred_2 <- predict(f_fit, new_data = lung2[1:2, ], type = "survival",
+  new_data_3 <- lung2[1:3, ]
+  f_pred <- predict(f_fit, new_data = new_data_3, type = "survival",
                     time = c(100, 200), penalty = 0.1)
 
-  expect_s3_class(pred_2, "tbl_df")
-  expect_equal(names(pred_2), ".pred")
-  expect_equal(nrow(pred_2), 2)
+  expect_s3_class(f_pred, "tbl_df")
+  expect_equal(names(f_pred), ".pred")
+  expect_equal(nrow(f_pred), nrow(new_data_3))
   expect_true(
-    all(purrr::map_lgl(pred_2$.pred, ~ all(dim(.x) == c(2, 2))))
+    all(purrr::map_lgl(f_pred$.pred, ~ all(dim(.x) == c(2, 2))))
   )
   expect_true(
-    all(purrr::map_lgl(pred_2$.pred,
+    all(purrr::map_lgl(f_pred$.pred,
                        ~ all(names(.x) == c(".time", ".pred_survival"))))
   )
 
+  exp_pred_multi <-
+    predict(f_fit, new_data = new_data_3, type = "survival",
+            time = c(100, 200), penalty = 0.05) %>%
+    tidyr::unnest(cols = .pred) %>%
+    dplyr::mutate(penalty = 0.05,
+                  .row = rep(1:3, each = 2)) %>%
+    dplyr::bind_rows(
+      f_pred %>%
+        tidyr::unnest(cols = .pred) %>%
+        dplyr::mutate(penalty = 0.1, .row = rep(1:3, each = 2))
+    ) %>%
+    dplyr::arrange(.row, .time) %>%
+    dplyr::select(penalty, .time, .pred_survival)
```

Comment:
I put this together from two separate calls to `predict()` with different penalty values for myself to check. Should this stay or change somehow?

Reply:
It was a little hard to follow, but I think the overall idea is right. I'd do:

```
f_pred2 <- predict(f_fit, ...)

f_pred_unnested <- f_pred %>% unnest() %>% mutate()
f_pred_unnested2 <- f_pred2 %>% unnest() %>% mutate()

exp_pred_multi <- bind_rows(f_pred_unnested, f_pred_unnested2) %>%
  arrange() %>%
  select()
```

Reply:
Random comment that I have a general aversion to piping into a `bind_rows()`. I think it is because most of the time all of the `bind_rows()` inputs are equally important, and a pipe normally implies that the first argument is more important than the others.

Reply:
right, that was a little messy :broom: